parser.y

```diff
@@ -10771,4 +10782,70 @@ LoadStatsStmt:
 		}
 	}
 
+/*******************************************************************************************/
+
+IndexAdviseStmt:
+	"INDEX" "ADVISE" LocalOpt "INFILE" stringLit MaxMinutes MaxIndexNum Lines
+	{
+		x := &ast.IndexAdviseStmt{
+			Path:	$5,
+		}
+		if $3 != nil {
+			x.IsLocal = true
+		}
+		if $6 != nil {
+			x.MaxMinutes = $6.(uint64)
+		}
+		if $7 != nil {
+			x.MaxIndexNum = $7.(*ast.MaxIndexNumClause)
+		}
+		if $8 != nil {
+			x.LinesInfo = $8.(*ast.LinesClause)
+		}
+		$$ = x
+	}
+
+MaxMinutes:
+	{
+		$$ = uint64(0)
+	}
```

Comment:
Can we take advantage of `types.UnspecifiedLength`? Thus we can distinguish 0 from an unspecified(default) number.

Reply:
There is indeed some ambiguity here, and using `types.UnspecifiedLength` is also a method to solve it.

But `types.UnspecifiedLength` is `-1`, which means the type of `MaxMinutes` needs to be changed from `uint` to `int`, it may cause unnecessary judgment.

Because in this scenario, **[-∞, 0]** is illegal input. If it is really necessary to specify a default value in *parser*, I think it is possible to define a constant similar to `types.UnspecifiedLength` to solve it, and the constant should be equal to `math.MaxUint64`.

Any suggestions? Or am I understanding it wrong?
(Same for the next few comments)

Reply:
Sure, you can create a similar constant using math.MaxUint64.

Reply:
Any suggestions for this one? 🤔

Reply:
IMHO, defining a constant named `UnspecifiedSize` is better.

Reply:
So should I also define it in the same position as `types.UnspecifiedLength`? In the types package

Reply:
How about putting it to `ast/util.go`?